lib/services/locale.dart

```diff
@@ -1,159 +1,243 @@
+    final cacheResponseContainerKey = 'countries';
+
+    final res = await client.call(
+      HttpMethod.get,
+      path: path,
+      params: params,
+      headers: headers,
+      cacheModel: cacheModel,
+      cacheKey: cacheKey,
+      cacheResponseIdKey: cacheResponseIdKey,
+      cacheResponseContainerKey: cacheResponseContainerKey,
+    );
+
+    return models.CountryList.fromMap(res.data);
+  }
+
+  /// List EU Countries
+  ///
+  /// List of all countries that are currently members of the EU. You can use the
+  /// locale header to get the data in a supported language.
+  ///
+  Future<models.CountryList> listCountriesEU() async {
+    const String path = '/locale/countries/eu';
+
+    final Map<String, dynamic> params = {};
+
+    final Map<String, String> headers = {
+      'content-type': 'application/json',
+    };
+
+    final cacheModel = '/locale/countries/eu';
+    final cacheKey = '';
+    final cacheResponseIdKey = 'code';
+    final cacheResponseContainerKey = 'countries';
+
+    final res = await client.call(
+      HttpMethod.get,
+      path: path,
+      params: params,
+      headers: headers,
+      cacheModel: cacheModel,
+      cacheKey: cacheKey,
+      cacheResponseIdKey: cacheResponseIdKey,
+      cacheResponseContainerKey: cacheResponseContainerKey,
+    );
+
+    return models.CountryList.fromMap(res.data);
+  }
+
+  /// List Countries Phone Codes
+  ///
+  /// List of all countries phone codes. You can use the locale header to get the
+  /// data in a supported language.
+  ///
+  Future<models.PhoneList> listCountriesPhones() async {
+    const String path = '/locale/countries/phones';
+
+    final Map<String, dynamic> params = {};
+
+    final Map<String, String> headers = {
+      'content-type': 'application/json',
+    };
+
+    final cacheModel = '/locale/countries/phones';
+    final cacheKey = '';
+    final cacheResponseIdKey = 'countryCode';
+    final cacheResponseContainerKey = 'phones';
+
+    final res = await client.call(
+      HttpMethod.get,
+      path: path,
+      params: params,
+      headers: headers,
+      cacheModel: cacheModel,
+      cacheKey: cacheKey,
+      cacheResponseIdKey: cacheResponseIdKey,
+      cacheResponseContainerKey: cacheResponseContainerKey,
+    );
+
+    return models.PhoneList.fromMap(res.data);
+  }
+
+  /// List Currencies
+  ///
+  /// List of all currencies, including currency symbol, name, plural, and
+  /// decimal digits for all major and minor currencies. You can use the locale
+  /// header to get the data in a supported language.
+  ///
+  Future<models.CurrencyList> listCurrencies() async {
+    const String path = '/locale/currencies';
+
+    final Map<String, dynamic> params = {};
+
+    final Map<String, String> headers = {
+      'content-type': 'application/json',
+    };
+
+    final cacheModel = '/locale/currencies';
+    final cacheKey = '';
+    final cacheResponseIdKey = 'code';
+    final cacheResponseContainerKey = 'currencies';
```

Comment:
@eldadfux, these are the additional labels we can add to help with caching:

| label | description |
| --- | --- |
| cacheModel | which local table to cache the data into |
| cacheKey | the key for the record in the cache |
| cacheResponseIdKey | the key in the response to use as the id for the record in the cache |
| cacheResponseContainerKey | the key in the response where the value is the list of records to cache |

Reply:
Do we need the 3 separate keys? Would it work with just 1 `cacheKey` that switches value depending on the route?

Reply:
I believe all 3 variables are needed sicne they're all used for different things.

1. `cacheKey` is the unique identifier for the record in the local cache. 
2. `cacheResponseIdKey` is the JSON parameter name for the ID of the record.
3. `cacheResponseContainerKey` is the JSON parameter name where the value is the list of records.